Close #4257: autodoc: Add autodoc_class_signature (#9171)
sphinx-doc/sphinx: doc/usage/extensions/autodoc.rst

   .. versionadded:: 1.4

.. confval:: autodoc_class_signature

>    This value selects what content will be shown as the class signature of an
>    :rst:dir:`autoclass` directive.  The possible values are:

Author: ```suggestion
   This value selects what will be shown as the class signature of an
   :rst:dir:`autoclass` directive.  The possible values are:
```

   ``"mixed"``

>       Display the signature with its class name.

Author: ```suggestion
      Display the signature after its class name.
```

   ``"separated"``

>       Separate the class entry and the definition of ``__init__()`` method

Author: ```suggestion
      Do not display the signature after the class name, and display the constructor as an independent method definition.
```
